src/tests/unit/cpu/nodes/reorder_node_test.cpp

```diff
@@ -82,185 +211,373 @@ class ReorderCustomImplTestBase: public ::testing::Test {
+        reorderNode->addEdge(childEdge);
+        auto rtParamsCache = std::make_shared<MKLDNNPlugin::MultiCache>(100);
+        inputDesc = srcBlockedDescCreator->createDesc(prec, srcDims);
+
+        const MKLDNNPlugin::CpuBlockedMemoryDesc outputDesc = dstBlockedDescCreator->createDesc(prec, dstDims);
+
+        auto parentMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+        auto childMemory = std::make_shared<MKLDNNPlugin::MKLDNNMemory>(cpuEngine);
+        parentMemory->Create(inputDesc, nullptr);
+        childMemory->Create(outputDesc, nullptr);
+        parentEdge->reuse(parentMemory);
+        childEdge->reuse(childMemory);
+
+        reorderNode->setDescs(inputDesc, outputDesc);
+
+        std::vector<std::shared_ptr<MKLDNNPlugin::MKLDNNNode>> nodes{inputNode, reorderNode, outputNode};
+        for (auto& n : nodes) {
+            n->init();
+            n->getSupportedDescriptors();
+            n->initSupportedPrimitiveDescriptors();
         }
+        //Select inputDesc as primitive descriptor for reorder node
+        reorderNode->selectPrimitiveDescriptorByIndex(0);
+        stream = {cpuEngine};
+        return;
     }
-    std::vector<uint8_t> dstData;
-    std::vector<uint8_t> srcData;
+
+private:
+    std::shared_ptr<MKLDNNPlugin::MultiCache> rtParamsCache;
+    mkldnn::stream stream;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNReorderNode> reorderNode;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNNode> inputNode;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNEdge> parentEdge;
+    std::shared_ptr<MKLDNNPlugin::MKLDNNEdge> childEdge;
+
+    MKLDNNPlugin::Shape srcDims;
+    MKLDNNPlugin::Shape dstDims;
+    LayoutType srcLayout;
+    LayoutType dstLayout;
+    InferenceEngine::Precision prec;
+    std::vector<std::vector<size_t>> inputShapes;
+    ngraph::PartialShape inputPartialShape;
+    BlockedDescCreator::CreatorsMap blockCreatorMap;
+    MKLDNNPlugin::CpuBlockedMemoryDesc inputDesc{InferenceEngine::Precision::FP32, MKLDNNPlugin::Shape{}};
 };
 
-TEST_P(ReorderNSPC2NCSPTest, NSPC2NCSP) {
+using ReorderCPUTestF32 = ReorderCPUTest<f32_f32>;
+using ReorderCPUTestS8 = ReorderCPUTest<s8_s8>;
+
+TEST_P(ReorderCPUTestF32, CompareResult) {
     Run();
 }
 
-TEST_P(ReorderNCSP2NSPCTest, NCSP2NSPC) {
+TEST_P(ReorderCPUTestS8, CompareResult) {
     Run();
 }
 
-const std::vector<bool> forceInplace {false, true};
-const auto NSPC2NCSPparams =::testing::Combine(
-                ::testing::Values(std::vector<size_t> {2, 16, 8, 8}),
-                ::testing::ValuesIn(forceInplace));
+const auto reorderCpuTestParams_1 =
+    ::testing::Values(ReorderCPUTestParamSetF32{{2, 16, 8, -1},
+                                                {{2, 16, 8, 8}, {2, 16, 8, 16}, {2, 16, 8, 8}},
+                                                LayoutType::nspc,
+                                                LayoutType::ncsp,
+                                                InferenceEngine::Precision::FP32});
```

Comment:
Can we have fully undefined shape {-1, -1, -1, -1} with target shapes with dimensions changing on all axis?